src/pages/Swap/index.tsx

```diff
@@ -196,14 +197,28 @@ export default function Swap() {
   const [approval, approveCallback] = useApproveCallbackFromTrade(trade /* allowedSlippage */)
 
   // check if user has gone through approval process, used to show two step buttons, reset on token change
-  const [approvalSubmitted, setApprovalSubmitted] = useState<boolean>(false)
+  const [approvalsSubmitted, setApprovalsSubmitted] = useState<boolean[]>([])
+
+  const currentTradeIndex = useMemo(() => {
+    if (allPlatformTrades) {
+      const index = allPlatformTrades.findIndex(
+        trade => trade && platformOverride && trade.platform.name === platformOverride.name
+      )
+      return index === -1 ? 0 : index
```

Comment:
I think we can simplify here :)

```suggestion
     return allPlatformTrades.findIndex(
        trade => trade && platformOverride && trade.platform.name === platformOverride.name
      ) || 0
```

Reply:
thats much smoother thank you!